---
title: "cancerGenomics"
author: "Pranathi Limmala"
date: '`r Sys.Date()`'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Overview
In this module, a Bioconductor tutorial for the Maftools package was followed to gain experience in analyzing cancer genomics data using the Mutation Annotation Format (MAF) file format. The MAF file, containing aggregated mutation information from VCF files generated on a project level, was used to summarize, analyze, annotate and visualize detected variants. The TCGA Project dataset was utilized as an example dataset. It is important to note that only true somatic mutations should be included in the MAF file used for analysis, after the removal of germline mutations and false positives. The resulting analysis using Maftools provided valuable insights into the genetic changes present in the dataset.

# This code chunk reads a MAF (Mutation Annotation Format) file for the TCGA LAML (Acute Myeloid Leukemia) dataset using the 'read.maf' function from the 'maftools' package. It also reads clinical information, such as survival and histology, from a TSV (Tab Separated Values) file, which is optional. The resulting 'laml' object contains the mutation data and clinical information in a format that can be further processed and visualized using other functions provided by the 'maftools' package. 
```{r}
library(maftools)
#path to TCGA LAML MAF file
laml.maf = system.file('extdata', 'tcga_laml.maf.gz', package = 'maftools') 
#clinical information containing survival information and histology. This is optional
laml.clin = system.file('extdata', 'tcga_laml_annot.tsv', package = 'maftools')
laml = read.maf(maf = laml.maf, clinicalData = laml.clin)
```

# This code chunk provides functions to explore and summarize the mutation data stored in the 'laml' object, which was created in the previous code block. The code provides various functions to extract sample-level, gene-level, and clinical data summaries, and lists all the fields (columns) in the MAF file. The 'write.mafSummary' function writes a summary of the MAF file to an output file with the specified basename. These functions can be used to gain an initial overview of the mutation data and select samples, genes, or mutations for further analysis.
```{r}
# 6.3 MAF object
#Typing laml shows basic summary of MAF file.
laml
#Shows sample summry.
getSampleSummary(laml)
#Shows gene summary.
getGeneSummary(laml)
#shows clinical data associated with samples
getClinicalData(laml)
#Shows all fields in MAF
getFields(x=laml)
#Writes maf summary to an output file with basename laml.
write.mafSummary(maf = laml, basename = 'laml')
```

# This chunk of code demonstrates various functions for visualizing mutation data in the 'laml' object using the 'maftools' package. 'plotmafSummary' produces a dashboard with an overview of the mutation data, while 'oncoplot' creates an oncoplot of the top mutated genes in the dataset. 'titv' and 'plotTiTv' functions generate summary statistics and plots for the transition/transversion ratio of the mutations, respectively. 'lollipopPlot' and 'plotProtein' create lollipop plots of amino acid changes and protein domains, respectively, for a specified gene. 'rainfallPlot' generates a rainfall plot of mutations in the 'brca' dataset, and 'tcgaCompare' compares the mutation load of the 'laml' dataset against other TCGA cohorts. These functions can help researchers to better understand and visualize the patterns of mutations in their datasets.
```{r}
# 7 Visualization
plotmafSummary(maf = laml, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)

# 7.2 Oncoplots
# 7.2.1 Drawing oncoplots
#oncoplot for top ten mutated genes.
oncoplot(maf = laml, top = 10)

# 7.3 Transition and Transversions
laml.titv = titv(maf = laml, plot = FALSE, useSyn = TRUE)
#plot titv summary
plotTiTv(res = laml.titv)

# 7.4 Lollipop plots for amino acid changes
#lollipop plot for DNMT3A, which is one of the most frequent mutated gene in Leukemia.
lollipopPlot(
  maf = laml,
  gene = 'DNMT3A',
  AACol = 'Protein_Change',
  showMutationRate = TRUE,
  labelPos = 882
)
plotProtein(gene = "TP53", refSeqID = "NM_000546")

# 7.5 Rainfall plots
brca <- system.file("extdata", "brca.maf.gz", package = "maftools")
brca = read.maf(maf = brca, verbose = FALSE)
rainfallPlot(maf = brca, detectChangePoints = TRUE, pointSize = 0.4)

# 7.6 Compare mutation load against TCGA cohorts
laml.mutload = tcgaCompare(maf = laml, cohortName = 'Example-LAML', logscale = TRUE, capture_size = 50)
```

# The code contains various functions and analyses performed on a MAF (Mutation Annotation Format) file for Acute Myeloid Leukemia (LAML) dataset.

#9.1 performs somatic interactions analysis on top 25 mutated genes based on their exclusive or co-occurring events.
#9.2 uses oncodrive algorithm to detect cancer driver genes based on positional clustering of mutations.
#9.3 adds and summarizes Pfam protein domains information to the MAF file.
#9.4 performs survival analysis based on mutation status of a given gene and predicts a set of genes associated with poor prognosis.
#9.5 compares two MAF files of primary and relapse Acute Promyelocytic Leukemia (APL) cohorts and plots forest plots, co-oncoplots, co-barplots, and lollipop plots to visualize the comparison.
#9.6 performs clinical enrichment analysis based on FAB (French-American-British) classification.
#9.7 predicts drug-gene interactions and provides a list of drugs interacting with DNMT3A.
#9.8 predicts oncogenic signaling pathways for the given MAF file and plots RTK-RAS pathway.
```{r}
# 9 Analysis
# 9.1 Somatic Interactions
#exclusive/co-occurance event analysis on top 10 mutated genes. 
somaticInteractions(maf = laml, top = 25, pvalue = c(0.05, 0.1))

# 9.2 Detecting cancer driver genes based on positional clustering
laml.sig = oncodrive(maf = laml, AACol = 'Protein_Change', minMut = 5, pvalMethod = 'zscore')
head(laml.sig)
plotOncodrive(res = laml.sig, fdrCutOff = 0.1, useFraction = TRUE, labelSize = 0.5)

# 9.3 Adding and summarizing pfam domains
laml.pfam = pfamDomains(maf = laml, AACol = 'Protein_Change', top = 10)

#Protein summary (Printing first 7 columns for display convenience)
laml.pfam$proteinSummary[,1:7, with = FALSE]

#Domain summary (Printing first 3 columns for display convenience)
laml.pfam$domainSummary[,1:3, with = FALSE]

# 9.4 Survival analysis
# 9.4.1 Mutation in any given genes
#Survival analysis based on grouping of DNMT3A mutation status
mafSurvival(maf = laml, genes = 'DNMT3A', time = 'days_to_last_followup', Status = 'Overall_Survival_Status', isTCGA = TRUE)

# 9.4.2 Predict genesets associated with survival
#Using top 20 mutated genes to identify a set of genes (of size 2) to predict poor prognostic groups
prog_geneset = survGroup(maf = laml, top = 20, geneSetSize = 2, time = "days_to_last_followup", Status = "Overall_Survival_Status", verbose = FALSE)
print(prog_geneset)
mafSurvGroup(maf = laml, geneSet = c("DNMT3A", "FLT3"), time = "days_to_last_followup", Status = "Overall_Survival_Status")

# 9.5 Comparing two cohorts (MAFs)
#Primary APL MAF
primary.apl = system.file("extdata", "APL_primary.maf.gz", package = "maftools")
primary.apl = read.maf(maf = primary.apl)
#Relapse APL MAF
relapse.apl = system.file("extdata", "APL_relapse.maf.gz", package = "maftools")
relapse.apl = read.maf(maf = relapse.apl)
#Considering only genes which are mutated in at-least in 5 samples in one of the cohort to avoid bias due to genes mutated in single sample.
pt.vs.rt <- mafCompare(m1 = primary.apl, m2 = relapse.apl, m1Name = 'Primary', m2Name = 'Relapse', minMut = 5)
print(pt.vs.rt)

# 9.5.1 Forest plots
forestPlot(mafCompareRes = pt.vs.rt, pVal = 0.1)

# 9.5.2 Co-onco plots
genes = c("PML", "RARA", "RUNX1", "ARID1B", "FLT3")
coOncoplot(m1 = primary.apl, m2 = relapse.apl, m1Name = 'PrimaryAPL', m2Name = 'RelapseAPL', genes = genes, removeNonMutated = TRUE)

# 9.5.3 Co-bar plots
coBarplot(m1 = primary.apl, m2 = relapse.apl, m1Name = "Primary", m2Name = "Relapse")

# 9.5.4 Lollipop plot-2
lollipopPlot2(m1 = primary.apl, m2 = relapse.apl, gene = "PML", AACol1 = "amino_acid_change", AACol2 = "amino_acid_change", m1_name = "Primary", m2_name = "Relapse")

#9.6 Clinical enrichment analysis
fab.ce = clinicalEnrichment(maf = laml, clinicalFeature = 'FAB_classification')
#Results are returned as a list. Significant associations p-value < 0.05
fab.ce$groupwise_comparision[p_value < 0.05]
plotEnrichmentResults(enrich_res = fab.ce, pVal = 0.05, geneFontSize = 0.5, annoFontSize = 0.6)

# 9.7 Drug-Gene Interactions
dgi = drugInteractions(maf = laml, fontSize = 0.75)
dnmt3a.dgi = drugInteractions(genes = "DNMT3A", drugs = TRUE)
#Printing selected columns.
dnmt3a.dgi[,.(Gene, interaction_types, drug_name, drug_claim_name)]

# 9.8 Oncogenic Signaling Pathways
OncogenicPathways(maf = laml)
PlotOncogenicPathways(maf = laml, pathways = "RTK-RAS")
```

# References:
https://www.ncbi.nlm.nih.gov/pubmed/?term=30341162
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3767041
https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format
https://northeastern.instructure.com/courses/135893/pages/binf6309-module-11-practice-cancer-genomics?module_item_id=8539194
